drivers/video/ov5640.c

```diff
@@ -0,0 +1,670 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ovti_ov5640
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/video.h>
+#include <zephyr/kernel.h>
+
+#define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ov5640);
+
+#include <zephyr/sys/byteorder.h>
+
+#define CHIP_ID_REG 0x300a
+#define CHIP_ID_VAL 0x5640
+
+#define SYS_CTRL0_REG     0x3008
+#define SYS_CTRL0_SW_PWDN 0x42
+#define SYS_CTRL0_SW_PWUP 0x02
+#define SYS_CTRL0_SW_RST  0x82
+
+#define IO_MIPI_CTRL00_REG 0x300e
+
+#define SC_PLL_CTRL0_REG     0x3034
+#define SC_PLL_CTRL1_REG     0x3035
+#define SC_PLL_CTRL2_REG     0x3036
+#define SC_PLL_CTRL3_REG     0x3037
+#define SYS_ROOT_DIVIDER_REG 0x3108
+#define PCLK_PERIOD_REG      0x4837
+
+#define DEFAULT_MIPI_CHANNEL 0
+
+#define OV5640_RESOLUTION_PARAM_NUM 24
+
+struct ov5640_config {
+	struct i2c_dt_spec i2c;
+#if DT_INST_NODE_HAS_PROP(0, reset_gpios)
+	struct gpio_dt_spec reset_gpio;
+#endif
+#if DT_INST_NODE_HAS_PROP(0, powerdown_gpios)
+	struct gpio_dt_spec powerdown_gpio;
+#endif
+};
+
+struct ov5640_data {
+	struct video_format fmt;
+};
+
+struct ov5640_reg {
+	uint16_t addr;
+	uint8_t val;
+};
+
+struct ov5640_mipi_clock_config {
+	uint8_t pllCtrl1;
+	uint8_t pllCtrl2;
+};
+
+struct ov5640_resolution_config {
+	uint16_t width;
+	uint16_t height;
+	const struct ov5640_reg *res_params;
+	const struct ov5640_mipi_clock_config mipi_pclk;
+};
+
+static const struct ov5640_reg ov5640InitParams[] = {
+	/* Power down */
+	{0x3008, 0x42},
+
+	/* System setting. */
+	{0x3103, 0x13},
```

Comment:
It looks like there is at least *limited* documentation on these registers present online: https://cdn.sparkfun.com/datasheets/Sensors/LightImaging/OV5640_datasheet.pdf. Can we please add `#define` definitions for these registers where possible?